app/components/container/containerController.js

```diff
@@ -196,6 +196,57 @@ function ($scope, $state, $stateParams, $filter, Container, ContainerCommit, Con
     });
   };
 
+  $scope.recreate = function() {
+    var config = ContainerHelper.configFromContainer($scope.container);
+    ModalService.confirm({
+      title: 'Are you sure ?',
+      message: 'You\'re about to re-create this container, any non-persisted data will be lost. This container will be removed and another one will be created using the same configuration.',
+      buttons: {
+        confirm: {
+          label: 'Recreate',
+          className: 'btn-danger'
+        }
+      },
+      callback: function onConfirm(confirmed) {
+        if(!confirmed) { return; }
+        else {
+          $('#loadingViewSpinner').show();
+          var container = $scope.container;
+          ContainerService.remove(container, true)
+          .then(function success(data) {
+            return ContainerService.createAndStartContainer(config);
```

Comment:
The first feature requested in #466 is to be able to re-create a container by pulling the latest image in the before re-creating it.

Please use the following promise chaining to be able to do that (you'll need to add the dependencies to `RegistryService` and `ImageService` in the controller definition):

```javascript
  .then(function success() {
    return RegistryService.retrieveRegistryFromRepository(container.Config.Image);
  })
  .then(function success(data) {
    return ImageService.pullImage(container.Config.Image, data, true);
  })
  .then(function success() {
    return ContainerService.createAndStartContainer(config);
  })
```